libs/sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -321,6 +321,8 @@ public Node visitFetch(FetchContext ctx) {
             scrollMode = Fetch.ScrollMode.RELATIVE;
             if (direction.ALL() != null) {
                 count = Long.MAX_VALUE;
+            } else if (direction.integerLiteral() != null) {
+                count = Long.parseLong(direction.integerLiteral().getText());
```

Comment:
Shall we add a parser test for this case as well?

Reply:
Good call, done